subseas_flex_fcst/maproom.py

```diff
@@ -0,0 +1,516 @@
+import os
+import flask
+import dash
+from dash import html
+import dash_bootstrap_components as dbc
+from dash.dependencies import Output, Input, State
+import dash_leaflet as dlf
+from pathlib import Path
+import pyaconf
+import pingrid
+import layout
+import plotly.graph_objects as pgo
+import plotly.express as px
+import numpy as np
+import cptio
+import xarray as xr
+from scipy.stats import t, norm, rankdata
+import pandas as pd
+
+CONFIG = pyaconf.load(os.environ["CONFIG"])
+
+PFX = CONFIG["core_path"]
+TILE_PFX = CONFIG["tile_path"]
+ADMIN_PFX = CONFIG["admin_path"]
+DATA_PATH = CONFIG["results_path"]
+
+# App
+
+CONFIG = pyaconf.load(os.environ["CONFIG"])
+
+SERVER = flask.Flask(__name__)
+APP = dash.Dash(
+    __name__,
+    server=SERVER,
+    external_stylesheets=[
+        dbc.themes.BOOTSTRAP,
+        "https://use.fontawesome.com/releases/v5.12.1/css/all.css",
+    ],
+    url_base_pathname=f"{PFX}/",
+    meta_tags=[
+        {"name": "description", "content": "Seasonal Forecast"},
+        {"name": "viewport", "content": "width=device-width, initial-scale=1.0"},
+    ],
+)
+APP.title = "Sub-Seasonal Forecast"
+
+APP.layout = layout.app_layout()
+
+
+@APP.callback(
+    Output("navbar-collapse", "is_open"),
+    Input("navbar-toggler", "n_clicks"),
+    State("navbar-collapse", "is_open"),
+)
+def toggle_navbar_collapse(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+
+@APP.callback(
+    Output("map", "center"),
+    Output("latInput", "min"),
+    Output("latInput", "max"),
+    Output("lngInput", "min"),
+    Output("lngInput", "max"),
+    Output("latLab", "children"),
+    Output("lonLab", "children"),
+    Input("submitLatLng","n_clicks"),
+)
+def initialize(toto):
+    fcst_mu = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_mu_file"])))
+    center_of_the_map = [((fcst_mu.Y[0]+fcst_mu.Y[-1])/2).values, ((fcst_mu.X[0]+fcst_mu.X[-1])/2).values]
+    lat_res = np.around((fcst_mu.Y[0]-fcst_mu.Y[1]).values, decimals=10)
```

Comment:
I am not sure... that might come from a Maproom usign ENACTS that has resolution 0.0375 and where grid centers are not on very round locations, making lat/lon_min/max with quite some decimals. Maybe that was in Senegal. I wonder if I decided to go even beyond so that it works for other typical high resolution we might use...
Let me find the commit elsewhere where I added this and see what I said then.

Reply:
Maybe we do one precision level up compared to data resolution for the toolkit pop up of the map?